llarp/router/router.hpp

```diff
@@ -242,15 +255,16 @@ namespace llarp
     NetworkConfig networkConfig;
     DnsConfig dnsConfig;
 
-    const IpAddress DefaultRPCBindAddr = IpAddress("127.0.0.1:1190");
+    const std::string DefaultRPCBindAddr = "tcp://127.0.0.1:1190";
     bool enableRPCServer = false;
-    std::unique_ptr<rpc::Server> rpcServer;
-    IpAddress rpcBindAddr = DefaultRPCBindAddr;
+    std::string rpcBindAddr = DefaultRPCBindAddr;
+    std::unique_ptr<rpc::RpcServer> m_RPCServer;
+
     const llarp_time_t _randomStartDelay;
 
-    /// lokid caller
-    std::unique_ptr<rpc::Caller> rpcCaller;
-    IpAddress lokidRPCAddr = IpAddress("127.0.0.1:22023");
+    std::shared_ptr<rpc::LokidRpcClient> m_lokidRpcClient;
+
+    std::string lokidRPCAddr = "ipc://loki.sock";
```

Comment:
Can be a lokimq::address.  This is going to open lock.sock in pwd, is that the right behaviour? (i.e. does the process chdir there?)

Reply:
Also lokimq::address has a factory function designed for this case:
```C++
auto lokidRPCAddr = lokimq::address::ipc("loki.sock");
```